api/v1alpha1/nats_types.go

```diff
@@ -81,6 +63,70 @@ type NATS struct {
 	Status NATSStatus `json:"status,omitempty"`
 }
 
+// NATSStatus defines the observed state of NATS.
+type NATSStatus struct {
+	State      State              `json:"state"`
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+// NATSSpec defines the desired state of NATS.
+type NATSSpec struct {
+	// Cluster defines configurations that are specific to NATS clusters.
+	Cluster Cluster `json:"cluster"`
+
+	// JetStream defines configurations that are specific to NATS JetStream.
+	// +optional
+	JetStream JetStream `json:"jetStream,omitempty"`
+
+	// JetStream defines configurations that are specific to NATS logging in NATS.
+	// +optional
+	Logging Logging `json:"logging,omitempty"`
```

Comment:
What about `resources`, `annotations` and `labels` in spec as proposed in the proposal?